soa-derive-internal/src/slice.rs

```diff
@@ -632,6 +661,64 @@ pub fn derive_mut(input: &Input) -> TokenStream {
                     #slice_from_raw_parts_mut
                 }
             }
+
+            #[doc(hidden)]
+            pub fn apply_permutation(&mut self, permutation: &mut soa_derive::Permutation) {
```

Comment:
I don't think this needs to be a public function. Everything should work fine with a private function!

Reply:
You are right! I didn't realize that it's essentially being called only from within the implementation of the generated code